src/NuGet.Core/NuGet.Packaging/Licenses/LicenseExpressionTokenizer.cs

```diff
@@ -0,0 +1,132 @@
+                if (!((_value[i] >= 'a' && _value[i] <= 'z') ||
+                    (_value[i] >= 'A' && _value[i] <= 'Z') ||
+                    (_value[i] >= '0' && _value[i] <= '9') ||
+                    _value[i] == ' ' ||
+                    _value[i] == '.' ||
+                    _value[i] == '-' ||
+                    _value[i] == '+' ||
+                    _value[i] == '(' ||
+                    _value[i] == ')'
+                    ))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Given a string, tokenizes by space into operators and values. The considered operators are, AND, OR, WITH, (, and ). 
+        /// </summary>
+        /// <returns>Tokens</returns>
+        internal IEnumerable<LicenseExpressionToken> Tokenize()
+        {
+            var potentialTokens = _value.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+            foreach (var token in potentialTokens)
+            {
+                var processingToken = token;
+
+                IList<LicenseExpressionToken> tokensAfterValue = null;
+
+                while (processingToken.Length > 0 && (processingToken[0] == '(' || processingToken[0] == ')'))
+                {
+                    yield return ParseBracket(processingToken[0]);
+
+                    processingToken = processingToken.Substring(1);
+                }
+
+                while (processingToken.Length > 0 && (processingToken[processingToken.Length - 1] == '(' || processingToken[processingToken.Length - 1] == ')'))
+                {
+                    if (tokensAfterValue == null)
+                    {
+                        tokensAfterValue = new List<LicenseExpressionToken>();
+
+                    }
+                    tokensAfterValue.Add(ParseBracket(processingToken[processingToken.Length - 1]));
+                    processingToken = processingToken.Substring(0, processingToken.Length - 1);
+                }
+
+                if (!string.IsNullOrEmpty(processingToken))
+                {
+                    yield return ParseTokenType(processingToken);
+                }
+
+                if (tokensAfterValue != null)
+                {
+                    foreach (var tokenAfterValue in tokensAfterValue)
+                    {
+                        yield return tokenAfterValue;
+                    }
+                }
+
+            }
+        }
+
+        private LicenseExpressionToken ParseBracket(char bracket)
+        {
+            if (bracket == '(')
+            {
+                return new LicenseExpressionToken(bracket.ToString(), LicenseTokenType.OPENING_BRACKET);
+            }
+            if (bracket == ')')
+            {
+                return new LicenseExpressionToken(bracket.ToString(), LicenseTokenType.CLOSING_BRACKET);
+            }
+            return null;
+        }
+
```

Comment:
Maybe add a comment to the method `<remarks>` saying that the method assumed brackets have already been parsed.